content/python/upgrade_virtual_guest_dedicated.py.md

```diff
@@ -0,0 +1,421 @@
+    {
+        "id": virtualGuestId
+    }
+]
+
+orderData = {
+    "complexType": "SoftLayer_Container_Product_Order_Virtual_Guest_Upgrade",
+    'packageId': package_id,
+    'prices': [{'id': prices[0],
+                'categories': [
+                    {
+                        'categoryCode': 'guest_disk1',
+                        'complexType': 'SoftLayer_Product_Item_Category'
+                    }
+                ],
+                },
+               {
+                   'id': prices[1],
+                   'categories': [
+                       {
+                           'categoryCode': 'guest_disk2',
+                           'complexType': 'SoftLayer_Product_Item_Category'
+                       }
+                   ],
+               }
+
+               ],
+    'properties': properties,
+    'virtualGuests': virtualGuests
+}
+
+orderService = client['SoftLayer_Product_Order']
+
+try:
+    response = orderService.verifyOrder(orderData)
+    print(json.dumps(response, sort_keys=False, indent=4, separators=(',', ': ')))
+
+except SoftLayer.SoftLayerAPIError as e:
+    """
+        If there was an error returned from the SoftLayer API then bomb out with the
+        error message.
+        """
+    print("Unable to upgrade the VSI \nfaultCode= %s, \nfaultString= %s"
+          % (e.faultCode, e.faultString))
+
+```
+
+Upgrade a VSI changing the size of a specific disk.
+```
+import json
+
+import SoftLayer
+
+client = SoftLayer.create_client_from_env()
+
+# The id of the virtual guest you wish upgrade
+virtualGuestId = 11111
+mask_prices = 'mask[prices]'
+package_keyname = 'CLOUD_SERVER'
+
+# items KeyNames to be upgrade
+item_keyName_list = ['GUEST_DISK_20_GB_SAN'] # "description": "20 GB (SAN)", "categoryCode": "guest_disk2"
+
+response_package = client["SoftLayer_Product_Package"].getAllObjects()
+package_id = 0
+for package in response_package:
+    if package.get('keyName') == package_keyname:
+        package_id = package.get('id')
+
+item_list = client["SoftLayer_Product_Package"].getItems(mask=mask_prices, id=package_id)
+
+prices = []
+for keyname_item in item_keyName_list:
+    for item in item_list:
+        if keyname_item == item.get('keyName'):
+            prices.append(item['prices'][0]['id'])
+
+print(' '.join(map(str, prices)))
+
+# Build a skeleton SoftLayer_Container_Product_Order_Property objects
+properties = [
+    {
+        "name": "MAINTENANCE_WINDOW",
+        "value": "2018-08-20T06:04:10Z"},
+    {
+        "name": "NOTE_GENERAL",
+        "value": "Upgrade instance configuration."
+    }
+]
+
+# Build a skeleton SoftLayer_Virtual_Guest object to model the id
+virtualGuests = [
+    {
+        "id": virtualGuestId
+    }
+]
+
+orderData = {
+    "complexType": "SoftLayer_Container_Product_Order_Virtual_Guest_Upgrade",
+    'packageId': package_id,
+    'prices': [{'id': prices[0],
+                'categories': [
+                    {
+                        'categoryCode': 'guest_disk1',
```

Comment:
At line 208 the comment indicates the disk is for `"categoryCode": "guest_disk2"` but here you are adding the disk to the `guest_disk1`

Reply:
Wrong comment, it is for the "categoryCode": "guest_disk1", I am upgrading the second disk of the vs from "GUEST_DISK_10_GB_SAN" to "GUEST_DISK_20_GB_SAN".